packages/fuels-core/src/utils.rs

```diff
@@ -1,13 +1,31 @@
 pub mod constants;
 pub mod offsets;
 
+use crate::error;
+use crate::types::errors::Result;
 use constants::{WITNESS_STATIC_SIZE, WORD_SIZE};
 use fuel_tx::Witness;
 
-pub const fn round_up_to_word_alignment(bytes_len: usize) -> usize {
-    (bytes_len + (WORD_SIZE - 1)) - ((bytes_len + (WORD_SIZE - 1)) % WORD_SIZE)
+pub fn checked_round_up_to_word_alignment(bytes_len: usize) -> Result<usize> {
+    let lhs = bytes_len.checked_add(WORD_SIZE - 1).ok_or_else(|| {
+        error!(
+            InvalidType,
+            "Addition overflow while rounding up {bytes_len} bytes to word alignment"
+        )
+    })?;
+    let rhs = lhs.checked_rem(WORD_SIZE).ok_or_else(|| {
+        error!(
+            InvalidType,
+            "Remainder overflow while rounding up {bytes_len} bytes to word alignment"
+        )
+    })?;
+    lhs.checked_sub(rhs).ok_or_else(|| {
+        error!(
+            InvalidType,
+            "Substraction overflow while rounding up {bytes_len} bytes to word alignment"
+        )
+    })
```

Comment:
I'm not sure this level of detail is useful to the user.

Maybe to be vague here, catch it somewhere higher up and give an error message like "Cannot handle xyz of this size"